### ISSUE TITLE
[Bug]: Some data containing many / leads to connection reset

### ISSUE THREAD

**Issue by IngoMaindorferEAWAG** [vernemq/vernemq#2349]
### Environment

- VerneMQ Version: 2.0.1
- OS: Debian 12
- Erlang/OTP version (if building from source):
- Cluster size/standalone: 2 Nodes

### Current Behavior

Hi,

I'm using Python with Paho to send data to my little cluster.

Trying to send this string `'base64_record: {base64_encoded: a=BhkuP0B3AAAAP6PWEEFaOyA/B0swQVsaQH/AAFwABgf8AAcH/AAMA/dv/QQamF8AAOEAA/o88QQVosID8HSwYZLj9OhwAAMEFbGkwABQf8AAYH/AAHwADAP3b90EGwo/////////////////////////////////////8=}}'`

leads to a connection reset.

If I reduces the string size by few chars, everything works (ommit /////////8=}}).

Transmitting the problematic string to HiveMQ works without a flaw,
Also no problems when I use mosquitto_pub connected to my Vernemq cluster.

So I'm not sure where to look.

A complete example:
```
import paho.mqtt.client as mqtt

def on_connect(mqttc, obj, flags, reason_code, properties):
    print("reason_code: " + str(reason_code))

def on_message(mqttc, obj, msg):
    print(msg.topic + " " + str(msg.qos) + " " + str(msg.payload))

def on_publish(mqttc, obj, mid, reason_code, properties):
    print("mid: " + str(mid))

def on_log(mqttc, obj, level, string):
    print(f"log: {string}")

def on_disconnect(client, userdata, disconnect_flags, reason_code, properties):
    print(f"disconnect: {reason_code}")

mqttc = mqtt.Client(mqtt.CallbackAPIVersion.VERSION2)
mqttc.on_message = on_message
mqttc.on_connect = on_connect
mqttc.on_publish = on_publish
mqttc.on_disconnect = on_disconnect
mqttc.on_log = on_log

mqttc.username_pw_set(username="MYUSER", password="SUPERSECRET")
mqttc.connect("MY_CLUSTER_IP", 1883, 60)

mqttc.loop_start()

json_string_broken = 'base64_record: {base64_encoded: a=BhkuP0B3AAAAP6PWEEFaOyA/B0swQVsaQH/AAFwABgf8AAcH/AAMA/dv/QQamF8AAOEAA/o88QQVosID8HSwYZLj9OhwAAMEFbGkwABQf8AAYH/AAHwADAP3b90EGwo/////////////////////////////////////8=}}'

json_string_ok = 'base64_record: {base64_encoded: a=BhkuP0B3AAAAP6PWEEFaOyA/B0swQVsaQH/AAFwABgf8AAcH/AAMA/dv/QQamF8AAOEAA/o88QQVosID8HSwYZLj9OhwAAMEFbGkwABQf8AAYH/AAHwADAP3b90EGwo////////////////}}'

info = mqttc.publish("sometopic", json_string_ok, qos=2)
info.wait_for_publish()

info = mqttc.publish("sometopic", json_string_broken, qos=2)
info.wait_for_publish()
```

### Expected behaviour

Data should be transmitted without connection getting closed.

### Code of Conduct

- [X] I agree to follow the VerneMQ's Code of Conduct

**Comment by ioolkos** [comment]
@IngoMaindorferEAWAG Thanks. There's a bunch of fixed and open protocol issues here https://github.com/vernemq/vernemq/issues/2283
But I guess what you see is not there. I'll try your provided script. Anything in the Verne logs related to this?

---
👉 Thank you for supporting VerneMQ: https://github.com/sponsors/vernemq
👉 Using the binary VerneMQ packages commercially (.deb/.rpm/Docker) requires a paid subscription.

**Comment by IngoMaindorferEAWAG** [comment]
Thanks for your fast answer,

logs are clean and empty and transfering random data chunks of 8k with above Python script works flawless.

I'm not sure is it a VerneMQ problem or a Paho one because transmitting this problematic string through mosquitto_pub to VerneMQ works...

But transferring the problematic string via Paho to HiveMQ works also, so I'm a bit confused...

Cheers,

Ingo

**Comment by mths1** [comment]
Hello @IngoMaindorferEAWAG 

I tried your code and it works perfectly fine for me. I have seen the following post where someone has a similar problem as you do:

https://stackoverflow.com/questions/66785852/paho-mqtt-client-failing-for-special-payloads-forward-slashes-when-using-openv

**Comment by ioolkos** [comment]
@mths1, @IngoMaindorferEAWAG  Gold! Thanks for digging that up. It immediately convices me that the issue must be something similar. Networks are funny. (but let's see and make sure)

---
👉 Thank you for supporting VerneMQ: https://github.com/sponsors/vernemq
👉 Using the binary VerneMQ packages commercially (.deb/.rpm/Docker) requires a paid subscription.

**Comment by IngoMaindorferEAWAG** [comment]
I was able to narrow it down further. 

Only connections from our institute network are affected, all other tests from outside work fine.

After discussing this with our network admin, it turned out that there is a stateful firewall for outgoing traffic that also looks at the packets.

There is a rule for a six-year-old CVE-2019-11779 of Mosquitto, but it doesn't actually match because I'm publishing and not subscribing...

If my packet contains too many slashes, it cheekily closes my socket.

I was then able to convince the admin that this only affects Mosquitto and has been fixed for six years, so he can safely remove the rule.

Thank you for your support and super-fast responses.
